index.js

```diff
@@ -108,7 +108,33 @@ function resizeImage (filePath, requestData, data, options, cb) {
                 var width, height;
                 var size = metadata.size;
 
-                if (options.scaleUp || (!options.scaleUp && ( size.width > requestData.width || size.height > requestData.height))) {
+
+                // This lets us use just a width or height if desired
+                if("height" in requestData){
+                    requestData.height = parseInt(requestData.height);
+                }
+
+                if("width" in requestData){
+                    requestData.width = parseInt(requestData.width);
+                }
+
+                var aspect = size.width / size.height;
+
+                if(typeof(requestData.height) !== "undefined" && typeof(requestData.width) === "undefined" ){
+                    // var aspect = size.width / size.height;
+                    requestData.width = aspect * requestData.height;
+                }
+
+                if(typeof(requestData.width) !== "undefined" && typeof(requestData.height) === "undefined" ){
+                    // var aspect = size.height / size.width;
+                    requestData.height =  requestData.width / aspect;
+                }
+                    // might be a good idea to add a maximum size param? Don't want to store people's 24mp images.
```

Comment:
We should check this at the request stage, I think - but yes, definitely a good idea. This whole module needs a rewrite as it's not the most modular in the world, but I'll get around to that :)